### ISSUE TITLE
NS5000 when checking if an event subscription was received.

### ISSUE THREAD

**Issue by LiTo96** [nsubstitute/NSubstitute.Analyzers#202]
Since Version [1.0.16](https://github.com/nsubstitute/NSubstitute.Analyzers/releases/tag/1.0.16) the analyzer rule [NS5000](https://github.com/nsubstitute/NSubstitute.Analyzers/blob/master/documentation/rules/NS5000.md) alerts code like:
`sub.Received(1).PropertyChanged += MyPropertyChangedHandler;`

ErrorMessage:
NS5000: Unused received check. To fix, make sure there is a call after "Received". Correct: "sub.Received().SomeCall();". Incorrect: "sub.Received();"

**Comment by tpodolak** [comment]
@dtchepak I am trying to add valid test case for this scenario and I came up with this
```c#
using NSubstitute;
using System;
using Xunit;

namespace MyNamespace
{
    public class Foo
    {
        public event Action SomeEvent;
    }

    public class FooTests
    {
        [Fact]
        public void Test()
        {
            var substitute = NSubstitute.Substitute.For<Foo>();
            substitute.Received().SomeEvent += Arg.Any<Action>();
        }
    }
}
```
However, this test passes, even though SomeEvent was not fired - according to [docs](https://nsubstitute.github.io/help/received-calls/)
 it is one of the way of asserting if event subscription was received. Is this valid test case in this case or there is regression in NSubstitute?

**Comment by dtchepak** [comment]
This is because the member is not virtual. Running multiple tests including this one gives me:

```
  Error Message:
   NSubstitute.Exceptions.RedundantArgumentMatcherException : Some argument specifications (e.g. Arg.Is, Arg.Any) were left over after the last call.
```

If it is `virtual` it fails as expected.

```
  Error Message:
   NSubstitute.Exceptions.ReceivedCallsException : Expected to receive a call matching:
        SomeEvent += any Action
Actually received no matching calls.
```

**Comment by tpodolak** [comment]
Ahh right. I dont know why I thought that events were handled differently. Will update my PR accordingly

**Comment by jm-atbas** [comment]
Can you release an update with this fix to NuGet, please?
